src/codegen/sys/lib_.rs

```diff
@@ -23,6 +23,26 @@ pub fn generate(env: &Env) {
     save_to_file(&path, env.config.make_backup, |w| generate_lib(w, env));
 }
 
+fn get_library_name(lib: &String) -> String {
+    let mut s = String::from(lib);
+
+    if s.starts_with("lib") {
+        s = s.split_off("lib".len());
+    }
+
+    let offset = s.find(".so").unwrap_or(s.len());
+    s.truncate(offset);
+    s
+}
+
+fn write_link_attr(w: &mut dyn Write, shared_libs: &Vec<String>) -> Result<()> {
+    for it in shared_libs.iter() {
```

Comment:
```suggestion
    for it in &shared_libs {
```

Reply:
I guess you meant `for it in shared_libs` :) done